todo-mvc/src/widgets/createTodoItem.ts

```diff
@@ -105,15 +143,15 @@ const createTodoItem: TodoItemFactory = createWidget
 				}
 				return todoItem.state.completed ? ['completed', ...classes] : classes;
 			},
-			getChildrenNodes(): VNode[] {
-				const todoItem = <TodoItem> this;
+			getChildrenNodes(this: TodoItem): VNode[] {
+				const { checkbox, label, button, editInput } = childrenMap.get(this);
 				return [
 					h('div.view', [
```

Comment:
**note to self:** checkbox/label/button should probably encapsulated in it's own container widget then we could do away with the cache and `getChildrenNodes` override....